src/connection_string.ts

```diff
@@ -184,8 +185,22 @@ const FALSEHOODS = new Set(['false', 'f', '0', 'n', 'no', '-1']);
 function getBoolean(name: string, value: unknown): boolean {
   if (typeof value === 'boolean') return value;
   const valueString = String(value).toLowerCase();
-  if (TRUTHS.has(valueString)) return true;
-  if (FALSEHOODS.has(valueString)) return false;
+  if (TRUTHS.has(valueString)) {
```

Comment:
The spec for parsing booleans says that while drivers are encouraged to support values other than `true/false`, a warning is supposed to be emitted: https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst#values